rmw_zenoh_cpp/src/detail/rmw_context_impl_s.cpp

```diff
@@ -0,0 +1,482 @@
+
+///=============================================================================
+std::string rmw_context_impl_s::enclave() const
+{
+  std::lock_guard<std::recursive_mutex> lock(data_->mutex_);
+  return data_->enclave_;
+}
+
+///=============================================================================
+z_session_t rmw_context_impl_s::session() const
+{
+  std::lock_guard<std::recursive_mutex> lock(data_->mutex_);
+  return z_loan(data_->session_);
+}
+
+///=============================================================================
+std::optional<zc_owned_shm_manager_t> & rmw_context_impl_s::shm_manager()
+{
+  std::lock_guard<std::recursive_mutex> lock(data_->mutex_);
+  return data_->shm_manager_;
+}
+
+///=============================================================================
+rmw_guard_condition_t * rmw_context_impl_s::graph_guard_condition()
+{
+  std::lock_guard<std::recursive_mutex> lock(data_->mutex_);
+  return data_->graph_guard_condition_;
+}
+
+///=============================================================================
+size_t rmw_context_impl_s::get_next_entity_id()
+{
+  std::lock_guard<std::recursive_mutex> lock(data_->mutex_);
+  return data_->next_entity_id_++;
+}
+
+///=============================================================================
+rmw_ret_t rmw_context_impl_s::shutdown()
+{
+  return data_->shutdown();
+}
+
+///=============================================================================
+bool rmw_context_impl_s::is_shutdown() const
+{
+  std::lock_guard<std::recursive_mutex> lock(data_->mutex_);
+  return data_->is_shutdown_;
+}
+
+///=============================================================================
+bool rmw_context_impl_s::session_is_valid() const
+{
+  std::lock_guard<std::recursive_mutex> lock(data_->mutex_);
+  return z_check(data_->session_);
+}
+
+///=============================================================================
+rmw_ret_t rmw_context_impl_s::get_node_names(
+  rcutils_string_array_t * node_names,
+  rcutils_string_array_t * node_namespaces,
+  rcutils_string_array_t * enclaves,
+  rcutils_allocator_t * allocator) const
+{
+  std::lock_guard<std::recursive_mutex> lock(data_->mutex_);
+  return data_->graph_cache_->get_node_names(
+    node_names,
+    node_namespaces,
+    enclaves,
+    allocator);
+}
+
+///=============================================================================
+rmw_ret_t rmw_context_impl_s::get_topic_names_and_types(
+  rcutils_allocator_t * allocator,
+  bool no_demangle,
+  rmw_names_and_types_t * topic_names_and_types) const
+{
+  std::lock_guard<std::recursive_mutex> lock(data_->mutex_);
+  return data_->graph_cache_->get_topic_names_and_types(
+    allocator,
+    no_demangle,
+    topic_names_and_types);
+}
+
+///=============================================================================
+rmw_ret_t rmw_context_impl_s::publisher_count_matched_subscriptions(
+  const rmw_publisher_t * publisher,
+  size_t * subscription_count)
```

Comment:
To be consistent with the upper variables defined with `std::size_t`
```suggestion
  std::size_t * subscription_count)
```

Reply:
The memory for this raw pointer is allocated by rcl and is of type `size_t`. See https://github.com/ros2/rmw/blob/ac4f9afccca3dc305f2b519fdb0fb9131250f2a5/rmw/include/rmw/rmw.h#L651

So we shouldn't mix `std::size_t` and `size_t`.